tuf/client/updater.py

```diff
@@ -1152,6 +1154,16 @@ def _update_root_metadata(self, current_root_metadata):
       # _update_metadata().
       self.consistent_snapshot = True
       self._update_metadata('root', DEFAULT_ROOT_UPPERLENGTH, version=version)
+      # Ensure that the role and key information of the top-level roles is the
+      # latest.  We do this whether or not Root needed to be updated, in order
+      # to ensure that, e.g., the entries in roledb for top-level roles are
+      # populated with expected keyid info so that roles can be validated.  In
+      # certain circumstances, top-level metadata might be missing because it
+      # was marked obsolete and deleted after a failed attempt, and thus we
+      # should refresh them here as a protective measure.  See Issue #736.
+      self._rebuild_key_and_role_db()
+      self.consistent_snapshot = \
+          self.metadata['current']['root']['consistent_snapshot']
```

Comment:
Comment in L1152-1154 claims that `self.consistent_snapshot` is set with `_update_metadata()`.

https://github.com/theupdateframework/tuf/blob/c75a2b7fe1eaabc88f82a5f0bdfc8bd25cab7ab1/tuf/client/updater.py#L1152-L1154

 Why set it here explicitly? Is the comment above inaccurate?

Reply:
IDK, it's possible.